metricbeat/module/system/service/service.go

```diff
@@ -125,17 +162,68 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 		if !isOpen {
 			return nil
 		}
+	}
+
+	err = m.reportInactiveInst(report)
+	if err != nil {
+		return errors.Wrap(err, "error reporting inactive services")
+	}
+
+	return nil
+}
+
+func (m *MetricSet) reportInactiveInst(r mb.ReporterV2) error {
+	for name, inst := range m.instanceTracker {
+		fmt.Printf("====checking unit: %s\n", name)
+		// If the unit has timed out, remove it
+		if time.Now().Sub(inst.lastReported) >= m.cfg.InstantiatedTimeout {
+			delete(m.instanceTracker, name)
+			continue
+		}
+		// If it was reported, move on
+		if inst.reportedThisPeriod {
+			fmt.Printf("%s has been reported\n", name)
+			newInst := m.instanceTracker[name]
+			newInst.reportedThisPeriod = false
+			m.instanceTracker[name] = newInst
+			continue
+		}
+
+		props, err := getProps(m.conn, inst.unit.Name)
+		if err != nil {
+			m.Logger().Errorf("error getting properties for %s: %s", inst.unit.Name, err)
+			continue
+		}
 
+		event, err := formProperties(inst.unit, props)
+		if err != nil {
+			m.Logger().Errorf("error forming their properties for unit %s: %s", inst.unit.Name, err)
+			continue
+		}
+		isOpen := r.Event(event)
+		if !isOpen {
+			return nil
+		}
```

Comment:
Code here is quite similar to the code that reports normal services, could some of this be reused?